async-nats/src/lib.rs

```diff
@@ -611,14 +621,17 @@ pub async fn connect_with_options<A: ToServerAddrs>(
         events_tx.clone(),
     )?;
 
-    let connection = connector.try_connect().await?;
+    let (info, connection) = connector.try_connect().await?;
+
+    let (info_sender, info_watcher) = tokio::sync::watch::channel(info);
```

Comment:
Aside nit, for sometime: when we had one, wasn't much point but as we're getting more senders tx, rx convention is starting to make more sense.